src/test/regress/expected/start_stop_metadata_sync.out

```diff
@@ -0,0 +1,244 @@
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+SELECT count(*) > 0 FROM pg_class WHERE relname LIKE 'reference_table__' AND relnamespace IN (SELECT oid FROM pg_namespace WHERE nspname = 'start_stop_metadata_sync');
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+\c - - - :master_port
+SET search_path TO "start_stop_metadata_sync";
+SELECT * FROM distributed_table_1;
+ col | b
+---------------------------------------------------------------------
+(0 rows)
+
+ALTER TABLE distributed_table_4 DROP COLUMN b;
+SELECT stop_metadata_sync_to_node('localhost', :worker_1_port);
+ stop_metadata_sync_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+\d+ distributed_table_4
+                   Table "start_stop_metadata_sync.distributed_table_4"
+ Column |  Type   | Collation | Nullable | Default | Storage | Stats target | Description
+---------------------------------------------------------------------
+ a      | integer |           | not null |         | plain   |              |
+Indexes:
+    "distributed_table_4_a_key" UNIQUE CONSTRAINT, btree (a)
+    "ind1" btree (a)
+
+SELECT * FROM test_view;
+ count
+---------------------------------------------------------------------
+     1
+(1 row)
+
+SELECT * FROM test_matview;
+ count
+---------------------------------------------------------------------
+     1
+(1 row)
+
+SELECT count(*) > 0 FROM pg_dist_node;
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+SELECT count(*) > 0 FROM pg_dist_shard;
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+SELECT count(*) > 0 FROM pg_class WHERE relname LIKE 'distributed_table__' AND relnamespace IN (SELECT oid FROM pg_namespace WHERE nspname = 'start_stop_metadata_sync');
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+SELECT count(*) > 0 FROM pg_class WHERE relname LIKE 'reference_table__' AND relnamespace IN (SELECT oid FROM pg_namespace WHERE nspname = 'start_stop_metadata_sync');
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+\c - - - :worker_1_port
+SET search_path TO "start_stop_metadata_sync";
+-- these should not be found because of stop_metadata_sync_to_node
+SELECT * FROM distributed_table_1;
+ERROR:  relation "distributed_table_1" does not exist
+\d+ distributed_table_4
+SELECT * FROM test_view;
+ERROR:  relation "test_view" does not exist
+SELECT * FROM test_matview;
+ERROR:  relation "test_matview" does not exist
+SELECT count(*) > 0 FROM pg_dist_node;
+ ?column?
+---------------------------------------------------------------------
+ f
+(1 row)
+
+SELECT count(*) > 0 FROM pg_dist_shard;
```

Comment:
should we also check `pg_dist_shard_placement` and ` pg_dist_local_group_id`?

I think what I'd do is to define a function like `check_metadata()` where inside the function ensure that all the metadata is sane. If so, it returns true, else it returns false (or throws error -- whichever is simple to implement.

 And, we can use that function more heavily like the below, and the tests becomes much easier to follow:

```
\c master_port
SELECT check_metadata();

stop_sync();
SELECT check_metadata();

\c worker_node
SELECT check_metadata();

\c master_node
drop column
start_Sync
SELECT check_metadata();

\c worker_node
SELECT check_metadata();

.. and whatever more test you want like this
```

In the function, you can check what you already check (e.g., pg_dist_partition/shard/pg_class) and you can also check `pg_dist_shard_placment`, pg_dist_node and pg_dist_local_groupid.

Reply:
Sound good, but what will we check here? Like, are these tables empty or not? Or a more complex check to make sure that the metadata is correct?

Reply:
I think more complex check to make sure that the metadata is correct for this specific test (e.g., move the tests that you already have into a function and return false if anything is unexpected). 

The name of the function could be something less generic like `check_metadata_for_stop_sync_test()` or such